### ISSUE TITLE
[Ascend]: Inconsistent File Order from os.listdir() Causes Training Loss Mismatch During Precision Alignment Experiments

### ISSUE THREAD

**Issue by Crystal-jiang** [ByteDance-Seed/VeOmni#330]
### Add a description

**Describe the Bug**
When the input data_path is a directory containing more than one dataset file, the current code in dataset.py uses os.listdir(data_path) to retrieve files and add them to the dataset list. However, during precision alignment experiments, we observed that os.listdir() does not guarantee lexicographical (dictionary) order of the returned filenames across different machines.
This inconsistent file order directly affects subsequent data loading during training: two machines will load datasets in different sequences, leading to inconsistent training data inputs and ultimately resulting in mismatched training loss values between the two machines.
<img width="2558" height="1500" alt="Image" src="https://github.com/user-attachments/assets/7374fb7b-32cd-47ce-87e9-f28608848c9f" />  

**Expected Behavior**
The dataset files under the target directory should be loaded in a consistent lexicographical order across all machines, ensuring identical data loading sequences and consistent training loss during precision alignment experiments.  

**Actual Behavior**
os.listdir() returns filenames in non-deterministic order (not guaranteed to be lexicographical) across different machines, leading to inconsistent data loading sequences and mismatched training loss.

**Additional Context**
This issue was discovered during precision alignment experiments, where consistent training conditions (including data loading order) are critical for validating model precision consistency across different computing environments.

**Comment by Coach257** [comment]
maybe sort after listdir

**Comment by Crystal-jiang** [comment]
> maybe sort after listdir

Plan to submit a PR to add a sort step after calling os.listdir()
